test_regress/t/t_hier_block_perf.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+# DESCRIPTION: Verilator: Verilog Test driver/expect definition
+#
+# Copyright 2025 by Wilson Snyder. This program is free software; you
+# can redistribute it and/or modify it under the terms of either the GNU
+# Lesser General Public License Version 3 or the Perl Artistic License
+# Version 2.0.
+# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+
+import vltest_bootstrap
+
+test.scenarios('vlt_all')
+test.init_benchmarksim()
+test.cycles = (int(test.benchmark) if test.benchmark else 1000000)
+test.sim_time = test.cycles * 10 + 1000
+THREADS = int(os.environ["SIM_THREADS"]) if "SIM_THREADS" in os.environ else 2
+
+test.compile(benchmarksim=1,
+             v_flags2=["+define+SIM_CYCLES=" + str(test.cycles), "--prof-exec", "--hierarchical"],
+             threads=(THREADS if test.vltmt else 1))
+
+test.execute(all_run_flags=[
+    "+verilator+prof+exec+start+2",
+    " +verilator+prof+exec+window+2",
+    " +verilator+prof+exec+file+" + test.obj_dir + "/profile_exec.dat",
+    " +verilator+prof+vlt+file+" + test.obj_dir + "/profile.vlt"])  # yapf:disable
+
+test.run(cmd=[
+    os.environ["VERILATOR_ROOT"] + "/bin/verilator_gantt", test.obj_dir +
+    "/profile_exec.dat", "--vcd " + test.obj_dir + "/profile_exec.vcd", "| tee " + test.obj_dir +
+    "/gantt.log"
+])
```

Comment:
I think this whole test will pass even if none of the costs get propagated?  Can we somehow make it fail if they don't?  Maybe add a V3Stat showing how many costs have propagated, then run with --stats and check that?   Or add a --debug-costs-exist flag that asserts if a cost is non-zero and use that?

(If I missed some check that does make this fail, just add a comment in this file as to the mechanism and we're all set.)

Reply:
I'd added stat for tracking how many hierarchical DPI prototypes have costs provided.